tests/build_tests/super_errors/expected/record_coercion_optional_mismatch.res.expected

```diff
@@ -0,0 +1,13 @@
+
+  [1;31mWe've found a bug for you![0m
+  [36m/.../fixtures/record_coercion_optional_mismatch.res[0m:[2m14:10-15[0m
+
+  12 [2m│[0m }
+  13 [2m│[0m 
+  [1;31m14[0m [2m│[0m let y = ([1;31mx :> y[0m)
+  15 [2m│[0m 
+
+  Type x is not a subtype of y 
+
+  The record [1;33mx[0m cannot be coerced to the record [1;33my[0m because:
+  - The field [1;33mx[0m is optional in record [1;33mx[0m, but is not optional in record [1;33my[0m
```

Comment:
the error message is very detailed and insightful, nice! That reminds me that the other way around is not yet possible, you can't coerce to a type that has more optional fields than the original type.

Reply:
Would that be reasonable to support?

Reply:
definitely low-priority but I once had the need for such a thing and I can't find of any argument against it.

Reply:
@cristianoc you see any issues with supporting it?